packages/portfolio-contract/tools/plan-solve.ts

```diff
@@ -495,6 +441,28 @@ export const rebalanceMinCostFlowSteps = async (
     }),
   );
 
+  // Validate flow consistency after scheduling (optional, only in debug mode)
+  if (graph.debug) {
+    const validation = validateSolvedFlows(graph, prioritized, FLOW_EPS);
+    if (!validation.ok) {
+      console.error('[solver] Flow validation failed:', validation.errors);
+      console.log('[solver] Original supplies:', graph.supplies);
+      console.log('[solver] Scheduling deadlock. Final supplies:', supplies);
+      console.log(
+        '[solver] All proposed flows in order:',
+        JSON.stringify(
+          steps,
+          (k, v) => (typeof v === 'bigint' ? v.toString() : v),
+          2,
+        ),
+      );
```

Comment:
```suggestion
      console.error('[solver] Flow validation failed:', validation.errors);
      console.error('[solver] Original supplies:', graph.supplies);
      console.error('[solver] Scheduling deadlock. Final supplies:', supplies);
      console.error('[solver] All proposed flows in order:', steps);
```
or if the latter is too noisy, maybe pull in [`readableSteps`](https://github.com/Agoric/agoric-sdk/blob/a61c7fe585d715bc448459e3de9ac76f7decc3be/packages/portfolio-contract/test/rebalance.test.ts#L63):
```suggestion
      console.error('[solver] Flow validation failed:', validation.errors);
      console.error('[solver] Original supplies:', graph.supplies);
      console.error('[solver] Scheduling deadlock. Final supplies:', supplies);
      console.error(
        '[solver] All proposed flows in order:',
        readableSteps(steps),
      );
```